pkg/controller/direct/containerattached/containerattachedcluster_mappings.go

```diff
@@ -0,0 +1,189 @@
+	})
+	/*NOTYET
+	// MISSING: AdminGroups
+	*/
+	return out
+}
+
+func AttachedOidcConfig_FromProto(mapCtx *direct.MapContext, in *pb.AttachedOidcConfig) *krm.AttachedOidcConfig {
+	if in == nil {
+		return nil
+	}
+	out := &krm.AttachedOidcConfig{}
+	out.IssuerURL = in.GetIssuerUrl()
+	out.Jwks = in.GetJwks()
+	return out
+}
+
+func AttachedOidcConfig_ToProto(mapCtx *direct.MapContext, in *krm.AttachedOidcConfig) *pb.AttachedOidcConfig {
+	if in == nil {
+		return nil
+	}
+	out := &pb.AttachedOidcConfig{}
+	out.IssuerUrl = in.IssuerURL
+	out.Jwks = in.Jwks
+	return out
+}
+
+func ContainerAttachedClusterSpec_FromProto(mapCtx *direct.MapContext, in *pb.AttachedCluster) *krm.ContainerAttachedClusterSpec {
+	if in == nil {
+		return nil
+	}
+	out := &krm.ContainerAttachedClusterSpec{}
+	out.Description = direct.LazyPtr(in.GetDescription())
+	out.OidcConfig = *AttachedOidcConfig_FromProto(mapCtx, in.GetOidcConfig())
+	out.PlatformVersion = in.GetPlatformVersion()
+	out.Distribution = in.GetDistribution()
+	out.Fleet = *Fleet_FromProto(mapCtx, in.GetFleet())
+	/*NOTYET
+	// MISSING: Etag
+	*/
+	out.Annotations = in.Annotations
+	out.LoggingConfig = LoggingConfig_FromProto(mapCtx, in.GetLoggingConfig())
+	out.Authorization = AttachedClustersAuthorization_FromProto(mapCtx, in.GetAuthorization())
+	out.MonitoringConfig = MonitoringConfig_FromProto(mapCtx, in.GetMonitoringConfig())
+	/*NOTYET
+	// MISSING: ProxyConfig
+	*/
+	out.BinaryAuthorization = BinaryAuthorization_FromProto(mapCtx, in.GetBinaryAuthorization())
+	return out
+}
+
+func ContainerAttachedClusterSpec_ToProto(mapCtx *direct.MapContext, in *krm.ContainerAttachedClusterSpec) *pb.AttachedCluster {
+	if in == nil {
+		return nil
+	}
+	out := &pb.AttachedCluster{}
+	out.Description = direct.ValueOf(in.Description)
+	out.OidcConfig = AttachedOidcConfig_ToProto(mapCtx, &in.OidcConfig)
+	out.PlatformVersion = in.PlatformVersion
+	out.Distribution = in.Distribution
+	out.Fleet = Fleet_ToProto(mapCtx, &in.Fleet)
+	/*NOTYET
+	// MISSING: Etag
+	*/
+	out.Annotations = in.Annotations
+	out.LoggingConfig = LoggingConfig_ToProto(mapCtx, in.LoggingConfig)
+	out.Authorization = AttachedClustersAuthorization_ToProto(mapCtx, in.Authorization)
+	out.MonitoringConfig = MonitoringConfig_ToProto(mapCtx, in.MonitoringConfig)
+	/*NOTYET
+	// MISSING: ProxyConfig
+	*/
+	out.BinaryAuthorization = BinaryAuthorization_ToProto(mapCtx, in.BinaryAuthorization)
+	return out
+}
+
+func LoggingComponentConfig_FromProto(mapCtx *direct.MapContext, in *pb.LoggingComponentConfig) *krm.LoggingComponentConfig {
+	if in == nil {
+		return nil
+	}
+	out := &krm.LoggingComponentConfig{}
+	out.EnableComponents = slice_FromProto_two(mapCtx, in.EnableComponents, func(mapCtx *direct.MapContext, p pb.LoggingComponentConfig_Component) *string {
+		return direct.Enum_FromProto(mapCtx, p)
+	})
+	return out
+}
+
+func slice_FromProto_two[T, U any](mapCtx *direct.MapContext, in []T, mapper func(mapCtx *direct.MapContext, in T) *U) []U {
```

Comment:
Is this because the current Slice_FromProto has a pointer in the mapping function? If so, I think it would be worth adding this in the `direct` package to be reused. That can happen as a follow on PR. Let's rename this to something like `slice_FromProto` for now if so.

Reply:
That is correct.
And I did intend to rename it because this name is silly, I just forgot. My mistake!